pkg/cli/util.go

```diff
@@ -0,0 +1,78 @@
+package cli
+
+import (
+	"net"
+
+	"github.com/pkg/errors"
+	"github.com/replicatedhq/kurl/pkg/netutils"
+	"github.com/spf13/cobra"
+)
+
+func newUtilCommand(cli CLI) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "util",
+		Short: "Utility commands",
+		PersistentPreRunE: func(cmd *cobra.Command, args []string) error {
+			return cli.GetViper().BindPFlags(cmd.PersistentFlags())
+		},
+		PreRunE: func(cmd *cobra.Command, args []string) error {
+			return cli.GetViper().BindPFlags(cmd.Flags())
+		},
+	}
+	return cmd
+}
+
+func newUtilIfaceFromIPCommand(cli CLI) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "iface-from-ip IP",
+		Short: "Gets the interface name for a given IP address",
+		Args:  cobra.ExactArgs(1),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			ip := net.ParseIP(args[0])
+			if ip.To16() == nil {
+				return errors.New("invalid IP address")
+			}
+			iface, err := netutils.GetInterfaceByIPv6(ip)
+			if err == nil {
+				cmd.Println(iface.Name)
+				return nil
+			}
+			if ip.To4() != nil {
+				iface, err := netutils.GetInterfaceByIP(ip)
+				if err != nil {
+					return err
+				}
+				cmd.Println(iface.Name)
+				return nil
+			}
+			return err
+		},
+	}
+	return cmd
+}
+
+func newUtilDefaultIfaceCommand(cli CLI) *cobra.Command {
+	ipv6 := false
+	cmd := &cobra.Command{
+		Use:   "default-gateway-iface",
+		Short: "Gets the default gateway interface name",
+		RunE: func(cmd *cobra.Command, args []string) error {
+			if ipv6 {
+				iface, err := netutils.GetDefaultV6GatewayInterface()
+				if err != nil {
+					return err
+				}
+				cmd.Println(iface.Name)
```

Comment:
do we really want this outputs or are they for debug only?
Why use cmd.Println?

Reply:
This is a cli. Output needs to be parseable so i dont think special characters are appropriate. to see how these are used see https://github.com/replicatedhq/kURL/pull/4029/files#diff-898fa6e3c189cf8168e2a7c50d5900083094a39b8ef1a28a9b4c82189d29abcaR28-R29